### DISCUSSION TITLE
I'm getting some errors while running fista

### DISCUSSION THREAD

**Discussion by stareae** [CERN/TIGRE#435] [Q&A]
Hi !

Thank you so much for developing such an excellent CT reconstruction learning toolkit. This helped me a lot in my process of learning CT  reconstruction. Now I'm learning the fista algorithm, but the reconstructed image was wrong.
1、I can't found the function named FISTA_mod;
2、the reconstructed imgFISTA got something wrong. All image pixels are NaN.
![2023-04-19 221322](https://user-images.githubusercontent.com/123911971/233102993-816752f6-f075-4638-ba4c-9d68bb86eafa.png).

I don't know if it's a problem caused by my bug.
So sorry to bother you, I would appreciate it if I could help with it.

**Comment by AnderBiguri** [comment]
Hey @stareae,

Fista mod is now part of Fista, perhaps we haven't updated the docs, can you share where is it mentioned?

Additionally, can you show a full code example that reproduces your error? Nans can come for many reasons, including your input data being nan, or a bug in the code.

**Reply by stareae** [reply]
Hello @AnderBiguri,

Thanks for the quick reply! 

I tried running the Demos file named d20_Algorithms05 about Fista. And the Fista_mod mentioned here.
![image](https://user-images.githubusercontent.com/123911971/233323783-6778fc65-7c3f-4bfd-9953-dbe1225a1bb5.png)
There were no errors during the run, but I intercepted a layer of the reconstructed image and found that its Voxels value was wrong.

I checked the input data and they are all correct. I'm trying debug to find out what cause this.

**Reply by AnderBiguri** [reply]
I think it's a bug then, demos should run well. If its not exactly the demo code that causes nans, I can only help if you provide the full code. I'll try to fix it ASAP.

**Reply by stareae** [reply]
In fact this is the code inside the demo, I got this error while running the demo.

**Reply by AnderBiguri** [reply]
I have now fixed some problems with it. 

I however have not fixed FISTA_mod results of NaNs. This is some numerical issue, and I think it has to do with "hyper" not being exactly the largest eigenvalue of A, which could be estimated with the power method #438 . Fixing that will probably fix the NaNs.

**Comment by AnderBiguri** [comment]
@stareae fantatic, easier to debug for me. I'll fix it ASAP.

**Reply by stareae** [reply]
Thank you very much！
